### PR TITLE
[triangle-wasm] Add new types for triangle-wasm

### PR DESCRIPTION
**Pull request by danikaze**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/triangle-wasm/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+// Type definitions for triangle-wasm 1.0
+// Project: https://github.com/brunoimbrizi/triangle-wasm (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/triangle-wasm/index.d.ts:2]: remove not needed template commment

File: types/triangle-wasm/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ * @param data
+ * @returns instance of `TriangulateIO`
+ */
+export function makeIO(data?: TriangulateData): TriangulateIO;
+
+/**
+ * Releases the allocated memory for an input/output instance.
+ *
+ * @param io reference to the stored i/o object
+ * @param all release all copied pointers
```
Comment (peterblazejewicz) [inline types/triangle-wasm/index.d.ts:100]:
```suggestion
 * @param [all] release all copied pointers
```

File: types/triangle-wasm/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    readonly numberofsegments?: number;
+    readonly numberofholes?: number;
+    readonly numberofregions?: number;
+    readonly numberofedges?: number;
+}
+
+/**
+ * Initialises the WASM module.
+ *
+ * @param path (default `'/'`) path to triangle.out.wasm (See [Distributing WASM](https://github.com/brunoimbrizi/triangle-wasm#distributing-wasm))
```
Comment (peterblazejewicz) [inline types/triangle-wasm/index.d.ts:66]:
```suggestion
 * @param [path] (default `'/'`) path to triangle.out.wasm (See [Distributing WASM](https://github.com/brunoimbrizi/triangle-wasm#distributing-wasm))
```

File: types/triangle-wasm/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    input: TriangulateIO,
+    output: TriangulateIO,
+    vorout?: TriangulateIO | null,
+): void;
+
+/**
+ * Creates an instance of `TriangulateIO` and allocates memory on the heap.
+ * `data` is only required when creating an input instance.
+ *
+ * @param data
```
Comment (peterblazejewicz) [inline types/triangle-wasm/index.d.ts:91]:
```suggestion
 * @param [data]
```

File: types/triangle-wasm/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+export function init(path?: string): Promise<void>;
+
+/**
+ * Triangulates the data passed in as `input` and writes the result to `ouput`
+ * (and the optional Voronoi result to `vorout`).
+ *
+ * @param switches
+ * @param input
+ * @param output
+ * @param vorout
```
Comment (peterblazejewicz) [inline types/triangle-wasm/index.d.ts:78]:
```suggestion
 * @param [vorout]
```

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Per other review

File: types/triangle-wasm/index.d.ts (hunk 2 above)
Comment (danikaze) [reply types/triangle-wasm/index.d.ts:100]: if this is for jsdocs optional parameters, shouldn't be the types enough for it?

**Review by RyanCavanaugh — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by danikaze** [comment]
Ready to merge

**Pull request merged**